src/dawn/CodeGen/GridTools/ASTStencilDesc.cpp

```diff
@@ -65,7 +65,53 @@ void ASTStencilDesc::visit(const std::shared_ptr<StencilCallDeclStmt>& stmt) {
 }
 
 void ASTStencilDesc::visit(const std::shared_ptr<BoundaryConditionDeclStmt>& stmt) {
-  DAWN_ASSERT_MSG(0, "BoundaryConditionDeclStmt not yet implemented");
+  if(instantiation_->getBoundaryConditionToExtentsMap().count(stmt) == 0) {
+    DAWN_ASSERT_MSG(false, "Boundary Condition did not trigger extent-calculation");
+  }
+  Extents extents = instantiation_->getBoundaryConditionToExtentsMap().find(stmt)->second;
+  int haloIMinus = extents[0].Minus;
+  int haloIPlus = extents[0].Plus;
+  int haloJMinus = extents[1].Minus;
+  int haloJPlus = extents[1].Plus;
+  int haloKMinus = extents[2].Minus;
+  int haloKPlus = extents[2].Plus;
+  std::string fieldname = stmt->getFields()[0]->Name;
+
+  // Set up the halos
+  std::string halosetup = dawn::format(
+      "gridtools::array< gridtools::halo_descriptor, 3 > halos;\n"
+      "halos[0] =gridtools::halo_descriptor(%i, %i, "
+      "%s->get_storage_info_ptr()->begin<0>(),%s->get_storage_info_ptr()->end<0>(), "
+      "%s->get_storage_info_ptr()->total_length<0>());\nhalos[1] = gridtools::halo_descriptor(%i, "
+      "%i, "
+      "%s->get_storage_info_ptr()->begin<1>(),%s->get_storage_info_ptr()->end<1>(), "
+      "%s->get_storage_info_ptr()->total_length<1>());\nhalos[2] = gridtools::halo_descriptor(%i, "
+      "%i, "
+      "%s->get_storage_info_ptr()->begin<2>(),%s->get_storage_info_ptr()->end<2>(), "
+      "%s->get_storage_info_ptr()->total_length<2>());\n",
+      haloIMinus, haloIPlus, fieldname, fieldname, fieldname, haloJMinus, haloJPlus, fieldname,
+      fieldname, fieldname, haloKMinus, haloKPlus, fieldname, fieldname, fieldname);
+  std::string makeView = "";
+
+  // Create the views for the fields
+  for(int i = 0; i < stmt->getFields().size(); ++i) {
+    auto fieldName = stmt->getFields()[i]->Name;
+    makeView +=
+        dawn::format("auto %s_view = GT_BACKEND_DECISION_viewmaker(%s);\n", fieldName, fieldName);
```

Comment:
where is this `viewmaker` coming from? is not GT code isnt? 
aside comment, where are creating views from several places, so it could make sense to have a function that in codegen that given the field name and backends, it returns the string to create a view... but you can take it for another PR if you prefer

Reply:
we take this to a new pr